src/main.h

```diff
@@ -52,6 +52,17 @@ CommandLineOptions;
 extern CommandLineOptions cl_options;
 
 
+/* Information about command line entries that can not be stored in GOptionEntry. */
+typedef struct
+{
+	gboolean persistent;  /* The option should be passed to "New Window" */
+}
+GeanyOptionEntryAux;
+
+extern GOptionEntry optentries[];
+extern GeanyOptionEntryAux optentries_aux[];
```

Comment:
We could add a function in libmain.c which could be called at argv parsing time and save a GPtrArray containing the `argv` to be forwarded into the already global `GeanyApp` structure. It would have the advantage of keeping all of the `argv`/GOption parsing code in the same file/place too.